### PR TITLE
fix: move wif functionality to PrivateKey

### PR DESCRIPTION
**Pull request by shruggr**

### CODE REVIEW

File: primitives/ec/privatekey.go [hunk 1]
```diff
@@ -41,12 +61,81 @@ func NewPrivateKey() (*PrivateKey, error) {
...
+
+	// For now, we assume compressed = true
+	// encodeLen := 1 + PrivateKeyBytesLen + 4
+	// if compress {
+	// 	encodeLen++
+	// }
+	encodeLen := 1 + PrivateKeyBytesLen + 4 + 1
+
+	a := make([]byte, 0, encodeLen)
+	a = append(a, net.PrivateKeyID)
```
Comment (sirdeggen) [inline primitives/ec/privatekey.go:124]: Does this mean we have to drag around chainParams just to encode to Wif? I wish we could have default value in here somehow. Assume mainnet, but allow override.

File: primitives/ec/wif_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+)
+
+func TestEncodeDecodeWIF(t *testing.T) {
+	mainWif := "Kxfd8ABTYZHBH3y1jToJ2AUJTMVbsNaqQsrkpo9gnnc1JXfBH8mn"
+	testWif := "cP2cb5BJycySSVSH7scRPUyN5ao1XpgXUv1DwDcCHuG1ZGkJQxzH"
+	privHex := "2b2afb5ea8d8c623acd6744547628988d86787003bb970afe15d471b727db79c"
+
+	mainPriv, _ := PrivateKeyFromWif(mainWif)
+	testPriv, _ := PrivateKeyFromWif(testWif)
+	assert.Equal(t, mainWif, mainPriv.Wif(&chaincfg.MainNet))
```
Comment (sirdeggen) [inline primitives/ec/wif_test.go:22]: This chaincfg.MainNet is a bit ugly. I recall thinking this stuff mattered when first using bsvjs - having the whole environment get set to a specific chain. Now I know it's literally just a prefix if you want to see it as a string, it feels like we can tuck it away somewhere.

File: primitives/ec/privatekey.go [hunk 3]
```diff
@@ -41,12 +61,81 @@ func NewPrivateKey() (*PrivateKey, error) {
...
+	}
+
+	// netID := decoded[0]
+	privKeyBytes := decoded[1 : 1+PrivateKeyBytesLen]
+	privKey, _ := PrivateKeyFromBytes(privKeyBytes)
+	return privKey, nil
+}
+
+func (p *PrivateKey) Wif(net *chaincfg.Params) string {
+	// Precalculate size.  Maximum number of bytes before base58 encoding
```
Comment (sirdeggen) [inline primitives/ec/privatekey.go:111]:
```suggestion
        if net == nil {
		net = &chaincfg.MainNet
	}
	// Precalculate size.  Maximum number of bytes before base58 encoding
```

**Review by sirdeggen — CHANGES_REQUESTED** [review]
Looks good, please add a default

**Comment by shruggr** [comment]
I've removed chaincfg and implemented a default mainnet functionality.

File: primitives/ec/privatekey.go (hunk 3 above)
Comment (sirdeggen) [reply primitives/ec/privatekey.go:121]: Would something like this work?

File: primitives/ec/privatekey.go [hunk 4]
```diff
@@ -1,15 +1,41 @@
...
+	// ErrMalformedPrivateKey describes an error where a WIF-encoded private
+	// key cannot be decoded due to being improperly formatted.  This may occur
+	// if the byte length is incorrect or an unexpected magic number was
+	// encountered.
+	ErrMalformedPrivateKey = errors.New("malformed private key")
+)
+
+type Network byte
+
+var (
```
Comment (shruggr) [inline primitives/ec/privatekey.go:30]: I've removed chaincfg from this code entirely

File: primitives/ec/privatekey.go [hunk 5]
```diff
@@ -41,12 +67,84 @@ func NewPrivateKey() (*PrivateKey, error) {
...
+		return nil, ErrChecksumMismatch
+	}
+
+	// netID := decoded[0]
+	privKeyBytes := decoded[1 : 1+PrivateKeyBytesLen]
+	privKey, _ := PrivateKeyFromBytes(privKeyBytes)
+	return privKey, nil
+}
+
+func (p *PrivateKey) Wif() string {
```
Comment (shruggr) [inline primitives/ec/privatekey.go:116]: Default to Mainnet

File: primitives/ec/privatekey.go [hunk 6]
```diff
@@ -41,12 +67,84 @@ func NewPrivateKey() (*PrivateKey, error) {
...
+	return p.WifPrefix(byte(MainNet))
+}
+
+func (p *PrivateKey) WifPrefix(prefix byte) string {
```
Comment (shruggr) [inline primitives/ec/privatekey.go:120]: Optional override

**Review by sirdeggen — APPROVED** [review]
Awesome thanks

**Pull request merged**